libs/unity/library/Runtime/Scripts/Media/MediaLine.cs

```diff
@@ -78,46 +56,86 @@ public IMediaTrackSource Source
                         UpdateSenderOnSourceChanged();
                         UpdateTransceiverDesiredDirection();
                     }
                 }
                 else
                 {
-                    throw new ArgumentException(nameof(Source) + " is not a MonoBehaviour component", nameof(Source));
+                    if (value.MediaKind != MediaKind)
+                    {
+                        throw new ArgumentException("Wrong media kind", nameof(Source));
+                    }
+                    if (!(value is IMediaTrackSourceInternal))
+                    {
+                        throw new ArgumentException("Missing interface IMediaTrackSourceInternal", nameof(Source));
+                    }
+                    if (value is MonoBehaviour mediaTrackSource)
+                    {
+                        if (_source != mediaTrackSource)
+                        {
+                            (_source as IMediaTrackSourceInternal)?.OnRemoveFromMediaLine(this);
+                            _source = mediaTrackSource;
+                            (_source as IMediaTrackSourceInternal).OnAddedToMediaLine(this);
+                            UpdateSenderOnSourceChanged();
+                            UpdateTransceiverDesiredDirection();
+                        }
+                    }
+                    else
+                    {
+                        throw new ArgumentException(nameof(Source) + " is not a MonoBehaviour component", nameof(Source));
+                    }
                 }
             }
         }
 
         /// <summary>
-        /// Media sender (track) bridging the <see cref="Source"/> with the underlying <see cref="Transceiver"/>.
-        /// 
-        /// The media track is automatically managed when <see cref="Source"/> changes or the transceiver direction
-        /// is renegotiated, as needed.
+        /// Name of the local media track this component will create when calling <see cref="StartCaptureAsync"/>.
+        /// If left empty, the implementation will generate a unique name for the track (generally a GUID).
         /// </summary>
-        public MediaSender Sender { get; private set; } = null;
+        /// <remarks>
+        /// This value must comply with the 'msid' attribute rules as defined in
+        /// https://tools.ietf.org/html/draft-ietf-mmusic-msid-05#section-2, which in
+        /// particular constraints the set of allowed characters to those allowed for a
+        /// 'token' element as specified in https://tools.ietf.org/html/rfc4566#page-43:
+        /// - Symbols [!#$%'*+-.^_`{|}~] and ampersand &amp;
+        /// - Alphanumerical characters [A-Za-z0-9]
+        /// 
+        /// Users can manually test if a string is a valid SDP token with the utility
+        /// method <see cref="SdpTokenAttribute.Validate(string, bool)"/>.
+        /// </remarks>
+        /// <seealso cref="SdpTokenAttribute.Validate(string, bool)"/>
```

Comment:
nit: can we test that is a valid SDP token on `set`?

Reply:
Hmm is this done when assigning `_senderTrackName`?

Reply:
`_senderTrackName` has the attribute, so this will make some error message appear when editing the field in the Unity editor if the SDP token is not valid. We can add some extra validation in the setter too for runtime.

Reply:
Fixed.